Added setting to collapse saved rows when loading Grid and Fluid fields; #4873

intoeetive opened the pull request:
Added setting to collapse saved rows when loading Grid and Fluid fields; #4873

intoeetive committed:
Added setting to collapse saved rows when loading Grid and Fluid fields; #4873

intoeetive committed:
fix setting check

Yulyaswan committed:
Merge branch '7.dev' into feature/7.x/collapse-rows

# Conflicts:
#	system/ee/ExpressionEngine/Addons/grid/ft.grid.php
#	system/ee/ExpressionEngine/View/_shared/table.php

Yulyaswan commented:
I think it needs a bit more work
1. when we have row numbers -> need more styles (this I can do)
<img width="1204" height="230" alt="Screenshot 2025-08-19 at 09 49 29" src="https://github.com/user-attachments/assets/2ed1595f-ec27-4633-90f5-d7a35bd4d3fc" />
2. Without row numbers, it hides grid rows, but basically, you can't do anything from Entry. No option to open and edit row. Unless it was the idea? @robinsowell (here we need to change the design to add a 'collapse' button somewhere)
<img width="1176" height="164" alt="Screenshot 2025-08-19 at 09 50 48" src="https://github.com/user-attachments/assets/f238e510-d1b7-4b72-b9f5-64c266a16a95" />
3. With Fluid field it works well because we have already option to collapse rows. and because of this we can change data from entry
<img width="1174" height="437" alt="Screenshot 2025-08-19 at 09 54 46" src="https://github.com/user-attachments/assets/0bf3ab97-e6bf-4cc4-b28d-65a7d7e4f23a" />

intoeetive committed:
fix colspan for "add new" row

show all cells in table view

intoeetive commented:
@Yulyaswan there has been an error in the logic here - the "collapse" setting is only intended to be used with Vertical view. I fixed that, however there are still few issues that maybe you could fix

1) the setting is shown initially, before you make any selection for layout type. I believe we already had fix for this in place? Not a big deal though

2) if you have the setting on, and also have Row Numbers on, the row numbers don't seem to show up

I also added a colspan fix for new row feature - it was breaking the columns now that you have row numbers. If this PR gets delayed maybe you'll want to include that separately

Yulyaswan committed:
Merge branch '7.dev' into feature/7.x/collapse-rows

Yulyaswan committed:
added styles for  collapsed row
